### PR TITLE
Create new GuiEvent HoverOnScene

### PR DESCRIPTION
**Pull request by ahcorde**
Signed-off-by: ahcorde <ahcorde@gmail.com>
# 🎉 New feature

## Summary

It publishes the onhover mouse 2D points

Required by: 
 - https://github.com/ignitionrobotics/ign-gazebo/pull/983

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [x] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [x] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge**

### CODE REVIEW

File: include/ignition/gui/GuiEvents.hh [hunk 1]
```diff
@@ -323,6 +323,29 @@ namespace ignition
...
       };
+
+      /// \brief Event which is called to broadcast the 2D coordinates of a
+      /// user's mouse hover within the scene.
+      class IGNITION_GUI_VISIBLE HoverOnScene : public QEvent
+      {
+        /// \brief Constructor
+        /// \param[in] _point The point at which the mouse is hovering within
+        /// the scene
+        public: explicit HoverOnScene(const math::Vector2i &_point);
```
Comment (chapulina) [inline include/ignition/gui/GuiEvents.hh:334]: How about using `common::MouseEvent` like we're doing on `(Right|Left)ClickOnScene`?

**Review by chapulina — CHANGES_REQUESTED** [review]

File: include/ignition/gui/GuiEvents.hh (hunk 1 above)
Comment (ahcorde) [reply include/ignition/gui/GuiEvents.hh:334]: @chapulina 

The hover event in the QML is just providing these two values

https://github.com/ignitionrobotics/ign-gui/blob/6f50c9c2212705f83a78591de9f5943a54eca11f/src/plugins/minimal_scene/MinimalScene.qml#L42-L44

Another option is to extend the current `common::MouseEvent` to allow hover events:

https://github.com/ignitionrobotics/ign-common/blob/491d95588d0f5bcea4d4b3d4ee08b7a2ef982cb2/events/include/ignition/common/MouseEvent.hh#L54-L70

or `MOVE` is the same as hover ?

Comment (chapulina) [reply include/ignition/gui/GuiEvents.hh:334]: Yeah `MOVE` with false `Dragging` is a hover :wink:

Comment (chapulina) [reply include/ignition/gui/GuiEvents.hh:334]: I think the main advantage of `MouseEvent` is that we can add modifier keys to the event later as needed. For example, hover + Shift

File: include/ignition/gui/GuiEvents.hh [hunk 2]
```diff
@@ -323,6 +323,29 @@ namespace ignition
...
+
+        /// \brief Unique type for this event.
+        static const QEvent::Type kType = QEvent::Type(QEvent::MaxUser - 13);
```
Comment (jennuine) [inline include/ignition/gui/GuiEvents.hh:337]: Noob question, why `-13`? Can this be documented in the comments?

Comment (chapulina) [reply include/ignition/gui/GuiEvents.hh:337]: Good idea to comment it, because it keeps coming up :sweat_smile: 

tl;dr: `gazebo::gui:GuiEvents` count up, `gui::GuiEvents` count down.

See context here: https://github.com/ignitionrobotics/ign-gui/pull/70#discussion_r436873992

**Review by chapulina — APPROVED** [review]
